code/modules/surgery/organs/lungs.dm

```diff
@@ -323,6 +323,128 @@
+				if (20 to 100)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 150)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,2)
+				if (100 to 200)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 250)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,4)
+				if (200 to 400)
+					monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 250)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,8)
+				if (400 to INFINITY)
+					monoxide_reagent.accumilation = max(monoxide_reagent.accumilation, 450)
+					H.reagents.add_reagent(/datum/reagent/carbon_monoxide,16)
+		else
+			var/datum/reagent/carbon_monoxide/monoxide_reagent = H.reagents.has_reagent(/datum/reagent/carbon_monoxide)
+			if(monoxide_reagent)
+				monoxide_reagent.accumilation = min(monoxide_reagent.accumilation, 150)
+				monoxide_reagent.metabolization_rate = 10 //purges 10 per tick
+
+	//if you can read this, please tell me a good way to reduce repeating this piece of code
+	// Sulfur Dioxide
+		var/sulfur_dioxide_pp = PP(breath,GAS_SO2)
+		if (prob(sulfur_dioxide_pp))
+			to_chat(H, "<span class='alert'>It hurts to breath.</span>")
+		if (sulfur_dioxide_pp >40)
+			H.emote("gasp")
+			H.adjustOxyLoss(5)
+			if (prob(sulfur_dioxide_pp/2))
+				to_chat(H, "<span class='alert'>Your throat closes up!</span>")
+				H.silent = max(H.silent, 3)
+		else
+			H.adjustOxyLoss(round(sulfur_dioxide_pp/8))
+		gas_breathed = breath.get_moles(GAS_SO2)
+		if (gas_breathed > gas_stimulation_min)
+			H.reagents.add_reagent(/datum/reagent/sulfur_dioxide,1)
+
+	// Ozone
+		var/ozone_pp = PP(breath,GAS_O3)
+		if (prob(ozone_pp))
+			to_chat(H, "<span class='alert'>Your heart feels funny.</span>")
+		if (ozone_pp >40)
+			H.emote("gasp")
+			H.adjustOxyLoss(5)
+			if (prob(ozone_pp/2))
+				to_chat(H, "<span class='alert'>Your throat closes up!</span>")
+				H.silent = max(H.silent, 3)
+		gas_breathed = breath.get_moles(GAS_O3)
+		if (gas_breathed > gas_stimulation_min)
+			H.reagents.add_reagent(/datum/reagent/ozone,1)
+
+	// Ammonia
+		var/ammonia_pp = PP(breath,GAS_AMMONIA)
+		if (prob(ammonia_pp)*2)
+			to_chat(H, "<span class='alert'>Your lungs feel terrible!</span>")
+
+		if (ammonia_pp > 10)
+			H.emote("gasp")
+			H.adjustOxyLoss(5)
+			H.adjustOxyLoss(round(ammonia_pp/8))
+			if (prob(ammonia_pp/2))
+				to_chat(H, "<span class='alert'>Your throat burns!</span>")
+				H.silent = max(H.silent, 2)
+		else
+			H.adjustOxyLoss(round(ammonia_pp/8))
+		gas_breathed = breath.get_moles(GAS_AMMONIA)
+		if (gas_breathed > gas_stimulation_min)
+			if(prob(25))//unlike the chlorine reagent ammonia doesnt do lung damage do we handle it here instead
+				H.adjustOrganLoss(ORGAN_SLOT_LUNGS,2*1.6)
+			//ammonia is actually disposed of naturally by humans, but extremely poorly by non mammals, maybe we can make it toxic ONLY to certain species (plural) sometime?
```

Comment:
would be easy in a different pr. just add a flag to lungs if they can process it.